array_family/flex_ext.py

```diff
@@ -1239,10 +1239,10 @@ def map_centroids_to_reciprocal_space(
         panel_numbers = cctbx.array_family.flex.size_t(self["panel"])
 
         for i, expt in enumerate(experiments):
-            if "imageset_id" in self:
-                sel_expt = self["imageset_id"] == i
-            else:
+            if "id" in self:
                 sel_expt = self["id"] == i
+            else:
+                sel_expt = self["imageset_id"] == i
```

Comment:
Perhaps the following would solve your problem in a less destructive way?
```suggestion
            if not crystal_frame and "imageset_id" in self:
                sel_expt = self["imageset_id"] == i
            else:
                sel_expt = self["id"] == i
```

Reply:
Am doing a little bit of hunting around - I have a suspicion that there are better ways to fix this problem, as there are other derived questions from this (like: is this part of the reason why re-running indexing on multi-lattice data does not re-index all lattiecs)

Reply:
BTW I recognise that this would fix the _behaviour_ but I worry that it is not fixing the underlying problem.

Reply:
This change also broke indexing with > 1 lattice...